src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -151,79 +138,28 @@ public TimeoutPersister(string timeoutConnectionString, string timeoutDataTableN
-                    .Where(TableQuery.GenerateFilterCondition("OwningTimeoutManager", QueryComparisons.Equal, endpointName));
-            }
+                            TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual, now.ToString(partitionKeyScope))),
+                        TableOperators.And,
+                        TableQuery.GenerateFilterCondition("OwningTimeoutManager", QueryComparisons.Equal, endpointName))
+                );
 
             var timeoutDataEntities = await timeoutDataTable.ExecuteQueryAsync(query, take: 1000).ConfigureAwait(false);
-            var result = timeoutDataEntities.OrderBy(c => c.Time);
-
-            var allTimeouts = result.ToList();
-            if (allTimeouts.Count == 0)
-            {
-                return new TimeoutsChunk(new TimeoutsChunk.Timeout[0], now.AddSeconds(1));
-            }
 
-            var pastTimeouts = allTimeouts.Where(c => c.Time > startSlice && c.Time <= now).ToList();
-            var futureTimeouts = allTimeouts.Where(c => c.Time > now).ToList();
+            var timeouts = timeoutDataEntities.Where(c => c.Time <= now).OrderBy(c => c.Time).ToList();
 
-            if (lastSuccessfulReadEntity != null)
-            {
-                var catchingUp = lastSuccessfulRead.Value.AddSeconds(catchUpInterval);
-                lastSuccessfulRead = catchingUp > now ? now : catchingUp;
-                lastSuccessfulReadEntity.LastSuccessfullRead = lastSuccessfulRead.Value;
-            }
-
-            var future = futureTimeouts.SafeFirstOrDefault();
-            var nextTimeToRunQuery = lastSuccessfulRead ?? (future?.Time ?? now.AddSeconds(1));
-
-            var timeoutsChunk = new TimeoutsChunk(
-                pastTimeouts.Where(c => !string.IsNullOrEmpty(c.RowKey))
-                    .Select(c => new TimeoutsChunk.Timeout(c.RowKey, c.Time))
-                    .Distinct(new TimoutChunkComparer())
-                    .ToArray(),
-                nextTimeToRunQuery);
+            var dueTimeouts = timeouts.Where(c => !string.IsNullOrEmpty(c.RowKey))
+                .Select(c => new TimeoutsChunk.Timeout(c.RowKey, c.Time))
+                .Distinct(new TimoutChunkComparer())
+                .ToArray();
 
-            updateSuccessfulReadOperationForNextSpin = GetUpdateSuccessfulRead(lastSuccessfulReadEntity);
-            return timeoutsChunk;
-        }
-
-        async Task TryUpdateSuccessfulRead(CloudTable timeoutManagerDataTable)
-        {
-            if (updateSuccessfulReadOperationForNextSpin != null)
-            {
-                try
-                {
-                    await UpdateSuccessfulRead(timeoutManagerDataTable, updateSuccessfulReadOperationForNextSpin).ConfigureAwait(false);
-                }
-                finally
-                {
-                    updateSuccessfulReadOperationForNextSpin = null;
-                }
-            }
+            return new TimeoutsChunk(dueTimeouts, now.AddSeconds(1));
```

Comment:
I agree with @SeanFeldman 

I would suggest we use 1 minute. This is the [max timeout interval that the core will allow](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/DelayedDelivery/TimeoutManager/ExpiredTimeoutsPoller.cs#L124):

```// we cap the next retrieval to max 1 minute this will make sure that we trip the circuit breaker if we
                // loose connectivity to our storage. This will also make sure that timeouts added (during migration) direct to storage
                // will be picked up after at most 1 minute```

@SeanFeldman informed me that when a new timeout is added, [the next timeout interval is evaluated and will be shortened if the new timeout is sooner](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/DelayedDelivery/TimeoutManager/ExpiredTimeoutsPoller.cs#L50).

In this case, I suggest that the default if we do not find a future timeout, we should set the timeout to 1 minute in the future instead of 1 second.